### PR TITLE
copyrights.csv: Fix and check "License" and "Needs Updated" fields

### PR DESCRIPTION
**Pull request by pehjota**
Commit 97c8feb8ca3b (pull #7903) included a comma in the "Notes" field of four files, which instead of being enclosed in quotes, overflowed into the "Needs Update" field.  So if those files are updated, update_copyrights would clobber part of the notes.

The same commit specified for 58 of LordBob's portraits a license of "GNU GPL v2+;CC BY-SA 4.0".  I don't know if the semicolon is supposed to mean "and" or "or".

"And" isn't legally possible, because the GPL doesn't allow others to relicense under CC BY-SA.  "Or" appears incorrect, because LordBob [licensed][1] his portraits under "the GNU GPL" and I can't find any evidence of him also licensing them under CC BY-SA 4.0.

@Pentarctagon, is this license fix correct, or did LordBob relicense some (but not all) of his portraits somewhere I can't find?

Commit 1ecd4f4d599c (pull #8195) updated the "Date" field but didn't clear "Needs Update" after update_copyrights set "Needs Update" and "MD5" for data/core/images/units/monsters/raven/harbinger.png.

@doofus-01, is the row for data/core/images/units/monsters/raven/harbinger.png otherwise up-to-date?

Make update_copyrights check for and warn about "Needs Update" instead of clobbering it.  This should trigger CI failures if someone forgets to update a row or accidentally puts into the "Needs Update" field important information that shouldn't be clobbered on future updates.

And make it check for and warn about possibly invalid licenses like "GNU GPL v2+;CC BY-SA 4.0".

[1]: https://forums.wesnoth.org/viewtopic.php?p=329342#p329342

### CODE REVIEW

**Comment by Pentarctagon** [comment]
Honestly I don't remember anymore about the licenses.

**Comment by soliton-** [comment]
> The same commit specified for 58 of LordBob's portraits a license of "GNU GPL v2+;CC BY-SA 4.0". I don't know if the semicolon is supposed to mean "and" or "or".
> 
> "And" isn't legally possible, because the GPL doesn't allow others to relicense under CC BY-SA. "Or" appears incorrect, because LordBob [licensed](https://forums.wesnoth.org/viewtopic.php?p=329342#p329342) his portraits under "the GNU GPL" and I can't find any evidence of him also licensing them under CC BY-SA 4.0.
> 
> @Pentarctagon, is this license fix correct, or did LordBob relicense some (but not all) of his portraits somewhere I can't find?

We can hopefully just ask @Lordbob here or ping him on discord/forums.

File: update_copyrights [hunk 1]
```diff
@@ -26,6 +26,16 @@ import sys
 # [6] = current md5 hash
 ##
 
+##
+# Add new licenses to this list:
+# Beware that GNU GPL v2+ can't be relicensed to anything else (e.g. CC BY-SA 4.0) except by the copyright holder.
```
Comment (soliton-) [inline update_copyrights:31]: Why is GPLv2+ mentioned specifically? It's the same with any other license besides CC0 type of stuff?

Comment (pehjota) [reply update_copyrights:31]: That comment was indeed too brief to be accurate.  Right, other than licenses like WTFPL and 0BSD and waivers like CC0 and Unlicense, all licenses forbid relicensing the original work (e.g. "this permission notice shall be included in all copies").  What I meant was that some licenses (e.g. permissive licenses or CC BY-SA 4.0 -> GPL v3-only) allow modifications to be under other licenses.  I'll work on clarifying that.

Comment (pehjota) [reply update_copyrights]: Clarified: "Avoid things like 'GNU GPL v2+;CC BY-SA 4.0', unless you mean to dual license under either GNU GPL v2+ or CC BY-SA 4.0.  GNU GPL v2+ and CC BY-SA 4.0 (e.g.  a GNU GPL v2+ file with CC BY-SA 4.0 modifications) isn't legally possible."  Better?

**Comment by doofus-01** [comment]
> @doofus-01, is the row for data/core/images/units/monsters/raven/harbinger.png otherwise up-to-date?

Yeah, I must have copied when I meant to cut.

**Comment by Lordbob** [comment]
Hey crew, I saw @pehjota 's PM on the forums just tonight. I haven't given a thought to licensing concerns for a loooooong time, since most art I produced for Wesnoth that got subsequently used in the game has been released for close to a decade if not more. Honestly, I never intended for it to have a commercial life beyond helping further the game, so pick whichever license works best for you as long as it remains in the open source domain and somewhat protected from outright theft.

Although, even theft is increasing becoming less of a concern, since nowadays anyone with basic AI know-how can churn out a hundred similar portraits in a matter of minutes - I mean, there would barely be a point in stealing them as far as I can tell. But, I digress.

**Comment by Pentarctagon** [comment]
Thanks for answering, I'll just go ahead and merge this then. Also our current stance is to not use AI generated art anyway.

**Pull request merged**